stablehlo/transforms/VhloLegalizeToStablehlo.cpp

```diff
@@ -464,6 +464,198 @@ SpecialResult convertSpecial(const OpConversionPattern<VhloOpTy>& pattern,
+    if (isSplatArray(vhloOp.getPrecisionConfigAttr(),
+                     vhlo::PrecisionV1Attr::get(pattern.getContext(),
+                                                vhlo::PrecisionV1::DEFAULT)))
+      eraseAttrs(vhloAttrs, "precision_config");
+  }
+  if constexpr (std::is_same<VhloOpTy,
+                             vhlo::DynamicBroadcastInDimOpV1>::value) {
+    if (isEmptyTensor(vhloOp.getKnownExpandingDimensionsAttr()))
+      eraseAttrs(vhloAttrs, "known_expanding_dimensions");
+    if (isEmptyTensor(vhloOp.getKnownNonexpandingDimensionsAttr()))
+      eraseAttrs(vhloAttrs, "known_nonexpanding_dimensions");
+  }
+  if constexpr (std::is_same<VhloOpTy, vhlo::DynamicGatherOpV1>::value ||
+                std::is_same<VhloOpTy, vhlo::GatherOpV1>::value) {
+    if (isBoolean(vhloOp.getIndicesAreSortedAttr(), false))
+      eraseAttrs(vhloAttrs, "indices_are_sorted");
+  }
+  if constexpr (std::is_same<VhloOpTy, vhlo::InfeedOpV1>::value) {
+    if (isEmptyString(vhloOp.getInfeedConfig()))
+      eraseAttrs(vhloAttrs, "infeed_config");
+    if (isEmptyArray(vhloOp.getLayout())) eraseAttrs(vhloAttrs, "layout");
+  }
+  if constexpr (std::is_same<VhloOpTy, vhlo::OutfeedOpV1>::value) {
+    if (isEmptyString(vhloOp.getOutfeedConfig()))
+      eraseAttrs(vhloAttrs, "outfeed_config");
+  }
+  if constexpr (std::is_same<VhloOpTy, vhlo::RecvOpV1>::value ||
+                std::is_same<VhloOpTy, vhlo::SendOpV1>::value) {
+    if (isBoolean(vhloOp.getIsHostTransferAttr(), false))
+      eraseAttrs(vhloAttrs, "is_host_transfer");
+  }
+  if constexpr (std::is_same<VhloOpTy, vhlo::ReduceWindowOpV1>::value) {
+    if (isSplatTensor(pattern, vhloOp.getWindowStridesAttr(), 1l))
+      eraseAttrs(vhloAttrs, "window_strides");
+    if (isSplatTensor(pattern, vhloOp.getBaseDilationsAttr(), 1l))
+      eraseAttrs(vhloAttrs, "base_dilations");
+    if (isSplatTensor(pattern, vhloOp.getWindowDilationsAttr(), 1l))
+      eraseAttrs(vhloAttrs, "window_dilations");
+    if (isSplatTensor(pattern, vhloOp.getPaddingAttr(), 0l))
+      eraseAttrs(vhloAttrs, "padding");
+  }
+  if constexpr (std::is_same<VhloOpTy, vhlo::ScatterOpV1>::value) {
+    if (isBoolean(vhloOp.getIndicesAreSortedAttr(), false))
+      eraseAttrs(vhloAttrs, "indices_are_sorted");
+    if (isBoolean(vhloOp.getUniqueIndicesAttr(), false))
+      eraseAttrs(vhloAttrs, "unique_indices");
+  }
+  if constexpr (std::is_same<VhloOpTy, vhlo::SelectAndScatterOpV1>::value) {
+    if (isSplatTensor(pattern, vhloOp.getWindowStridesAttr(), 1l))
+      eraseAttrs(vhloAttrs, "window_strides");
+    if (isSplatTensor(pattern, vhloOp.getPaddingAttr(), 0l))
+      eraseAttrs(vhloAttrs, "padding");
+  }
+  if constexpr (std::is_same<VhloOpTy, vhlo::SortOpV1>::value) {
+    if (isInteger(vhloOp.getDimensionAttr(), -1))
+      eraseAttrs(vhloAttrs, "dimension");
+    if (isBoolean(vhloOp.getIsStableAttr(), false))
+      eraseAttrs(vhloAttrs, "is_stable");
+  }
+  return success();
```

Comment:
Currently this function only returns success I think. Do you envision this `TODO validate` to be when it starts to return failures?

Reply:
Yes, that's right.